### ISSUE TITLE
I2C epuck communication not responding

### ISSUE THREAD

**Issue by baryn** [gctronic/Pi-puck#1]
Hello,

I am working on the epuck2 with the Pi-puck extension and I have mounted either a Raspberry PI Zero W on top. The mode Selector is on position 10 as expected for Linux communication. The Pi-puck firmware has not been changed.
I wanted to run the provided test scripts to check that the connection with the robot from the computer is working before running the epuck2 ROS2 driver.
The i2c has been activated with armbian-config and after running 
`i2cdetect -y -l`   
we get the two available i2c ports
`i2c-1 i2c`

Then
`i2cdetect -y 1` gives the following table
  0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f
00: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
10: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
20: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
30: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
40: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
50: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
60: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
70: 70 -- -- -- -- -- -- -- 
I guess it matches the [figure](https://projects.gctronic.com/epuck2/wiki_images/i2c-buses.png) from the documentation where address 0x70 refers to the i2C Switcher and is accessible via i2c-1. So, everything seems in order.

However, the e-puck2/e-puck2_imu.py cannot connect to I2C. In this script, I don't get the idea behind the addresses:
`I2C_CHANNEL = 12`
`LEGACY_I2C_CHANNEL = 4`
Should we have an i2c-12 or i2c-4 activated ? Or does it pass through the I2C switcher somehow ?

In the e-puck2_test.py we can see `ROB_ADDR = 0x1F` but on the figure this address refers to the Main micro.

I have tried with the last e-puk2 firmware and with several RPI Zero and BPI-M2 Zero.

So, at the moment I am not able to get or send any data to the robot. I might have missed something obvious but cannot see it.

Cheers

**Comment by gctronic** [comment]
Hi,

you need to first enable the I2C channels in the switcher used to communicate with the pi-puck devices and with the robot and its sensors. Basically you need to send "0x03" to register "0x01" at address "0x70" on channel "i2c-1", the following command should work: `i2cset -f -y 1 0x70 0x01 0x03`.
This command will enable 2 additional i2c channels, in the figure you mentioned before they are i2c-3 (to communicate with devices on the pi-puck extension) and i2c-4 (to communicate with the robot and its sensors including the imu).

Anyway in our supported system (Raspbian) the configuration is already done so you have all channels available once the system is booted.

The e-puck2 ROS2 driver is very interesting, can you share some code/information about it? You can contact us directly if you want (https://www.gctronic.com/contact.php).

Best regards

**Comment by baryn** [comment]
Hi,

Thank you for sharing. Actually, after looking more deeply in the repo I found the config.txt file which seems necessary to fully enable the i2c communication. I am still not sure what parameters are allowing it. The default config.txt on the armbian OS is not sufficient, so I added yours and after rebooting and running `python i2c_enable.py` with parameter BOARD_I2C_CHANNEL set to 1, it worked on the RPI. The following addresses are now visible and we can see that they refer to the robot and sensors as expected:
![i2c](https://user-images.githubusercontent.com/12564948/234555078-cb2b4561-1043-42e6-aa06-44d196bd7dc2.png)
The rest of the tests are working and the ROS2 driver should be working too now.

I was initially working with the Banana PI BPI-M2 Zero instead of the RPI Zero W for performance reasons but the architecture is a bit different since it is currently running on an Armbian OS. There is no /boot/config.txt but a /boot/armbianEnv.txt instead. 

But just moments before your reply I was trying to figure out what parameters might have to be added to it in order to activate the i2c too. Adding the config.txt lines to it did not do anything more. 
So, after running your command `i2cset -f -y 0 0x70 0x01 0x03` (on i2c-0 for the BPI) the same addresses appeared ! So, it's all good to me, and I could test that the ros2 driver was working too. 

Regarding the e-puck2 ROS2 driver we adapted the cross compilation to the BPI and ROS2 Humble version.

Best regards
